chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -216,6 +208,58 @@ impl NetworkState {
         PartialEdgeInfo::new(&self.config.node_id(), peer1, nonce, &self.config.node_key)
     }
 
+    /// Connects peer with given TcpStream.
+    /// It will fail (and log) if we have too many connections already,
+    /// or if the peer drops the connection in the meantime.
+    fn spawn_peer_actor(
+        self: &Arc<Self>,
+        clock: &time::Clock,
+        stream: TcpStream,
+        stream_cfg: StreamConfig,
+    ) {
+        if let Err(err) = PeerActor::spawn(clock.clone(), stream, stream_cfg, None, self.clone()) {
+            tracing::info!(target:"network", ?err, "PeerActor::spawn()");
+        };
+    }
+
+    pub async fn spawn_inbound(self: &Arc<Self>, clock: &time::Clock, stream: TcpStream) {
+        self.spawn_peer_actor(clock, stream, StreamConfig::Inbound);
+    }
+
+    pub async fn spawn_outbound(self: Arc<Self>, clock: time::Clock, peer_info: PeerInfo) {
+        let addr = match peer_info.addr {
+            Some(addr) => addr,
+            None => {
+                warn!(target: "network", ?peer_info, "Trying to connect to peer with no public address");
+                return;
+            }
+        };
+        // The `connect` may take several minutes. This happens when the
+        // `SYN` packet for establishing a TCP connection gets silently
+        // dropped, in which case the default TCP timeout is applied. That's
+        // too long for us, so we shorten it to one second.
+        //
+        // Why exactly a second? It was hard-coded in a library we used
+        // before, so we keep it to preserve behavior. Removing the timeout
+        // completely was observed to break stuff for real on the testnet.
```

Comment:
I now realize why stuff was broken back than -- because we've `.wait`ed, so the whole network was blocked. So, remiving the timeout should be fine probably. But also, yeah, it makes sense to timeout here anyway.